src/measure_ops.jl

```diff
@@ -103,18 +103,89 @@ function eigenbasis(op::YGate)
 end
 
 function measure!(
-    postprocess::PostProcess,
+    postprocess::NoPostProcess,  # operator measuring does not allow removing bits or resetting vaules.
     op::AbstractBlock,
     reg::AbstractRegister,
     locs::AllLocs;
     kwargs...,
 )
     _check_msize(op, reg, locs)
     E, V = eigenbasis(op)
-    res = measure!(postprocess, ComputationalBasis(), reg |> V', locs; kwargs...)
-    res2 = measure!(postprocess, ComputationalBasis(), reg, locs; kwargs...)
-    postprocess isa NoPostProcess && apply!(reg, V)
-    diag(mat(E))[Int64.(res).+1]
+    res = measure!(postprocess, BlockedBasis(diag(mat(E))), reg |> V', locs; kwargs...)
+    apply!(reg, V)
+    return res
+end
+
+struct BlockedBasis{VT}
```

Comment:
add a docstring to this? what is blocked basis?

Reply:
I will add some comments to it, since this is not an interface that will be expose to users.